test/stage1/behavior/vector.zig

```diff
@@ -495,75 +495,109 @@ test "vector reduce operation" {
             var r = @reduce(op, @as(Vector(N, TX), x));
             switch (@typeInfo(TX)) {
                 .Int, .Bool => expectEqual(expected, r),
-                .Float => {
```

Comment:
`expectWithinEpsilon` is dumb as a brick, that's why you need this to deal with NaNs.
If you don't believe me try changing the expected value of some nan-returning reduction, you'll see that the test will pass with no problem at all.

Reply:
> If you don't believe me

It is impossible. :)

> you'll see that the test will pass with no problem at all.

Tests passed on my PC.

Reply:
Any test producing a NaN will pass as the comparisons evaluate to false, if you remove this lines the test is not checking anything.